### PR TITLE
Add Date Filter in Analysis List

### PR DESCRIPTION
**Pull request by subinasr**

### CODE REVIEW

File: src/redux/initial-state/dev-lang.json [hunk 1]
```diff
@@ -1276,7 +1276,7 @@
         "-75339790": "Entries analyzed",
         "-65511645": "Sources analyzed",
         "-70983044": "Add pillar analysis",
-        "-81726696": "All analyses",
+        "-81726696": "All analyses    {count}",
```
Comment (tnagorra) [inline src/redux/initial-state/dev-lang.json]:
```suggestion
        "-81726696": "All analyses {count}",
```

File: src/views/AnalysisModule/index.tsx [hunk 2]
```diff
@@ -72,20 +80,29 @@ function AnalysisModule(props: AnalysisModuleProps) {
     const [analysisCount, setAnalysisCount] = useState(0);
     const [analysisIdToDelete, setAnalysisIdToDelete] = useState<number | undefined>();
     const [analysisToEdit, setAnalysisToEdit] = useState();
+    const [filter, setFilter] = useState<Filter | undefined>(undefined);
+
+    const analysisQueryOptions = useMemo(() => {
+        const endDate = filter?.endDate ? new Date(filter?.endDate) : new Date();
+        endDate.setDate(endDate.getDate() + 1);
```
Comment (tnagorra) [inline src/views/AnalysisModule/index.tsx:87]: Please add a note why a day was added

File: src/views/AnalysisModule/index.tsx [hunk 3]
```diff
@@ -111,7 +128,7 @@ function AnalysisModule(props: AnalysisModuleProps) {
         },
     );
 
-    const handleAnalysisDeleteClick = useCallback((toDeleteKey) => {
+    const handleAnalysisToDelete = useCallback((toDeleteKey) => {
```
Comment (tnagorra) [inline src/views/AnalysisModule/index.tsx]: Why was this name changed?

File: src/views/AnalysisModule/styles.scss [hunk 4]
```diff
@@ -53,7 +53,20 @@
 
     .all-analyses {
         .header {
-            margin: var(--dui-spacing-large);
+            margin: var(--dui-spacing-medium)
+                var(--dui-spacing-extra-large);
```
Comment (tnagorra) [inline src/views/AnalysisModule/styles.scss]: This can be a single line statement.

File: src/views/AnalysisModule/index.tsx [hunk 5]
```diff
@@ -72,20 +80,30 @@ function AnalysisModule(props: AnalysisModuleProps) {
     const [analysisCount, setAnalysisCount] = useState(0);
     const [analysisIdToDelete, setAnalysisIdToDelete] = useState<number | undefined>();
     const [analysisToEdit, setAnalysisToEdit] = useState();
+    const [filter, setFilter] = useState<Filter | undefined>(undefined);
+
+    const analysisQueryOptions = useMemo(() => {
+        const endDate = filter?.endDate ? new Date(filter?.endDate) : new Date();
+        endDate.setDate(endDate.getDate() + 1);
+        // A day added to compensate the difference in timezone in client and server
```
Comment (tnagorra) [inline src/views/AnalysisModule/index.tsx]: This comment is not correct.

File: src/views/AnalysisModule/Analysis/AnalysisPillar/index.tsx [hunk 6]
```diff
@@ -17,7 +17,7 @@ import styles from './styles.scss';
 interface ComponentProps extends Omit<AnalysisPillars, 'id' | 'analysis'> {
     pillarId: AnalysisPillars['id'];
     onDelete: (value: number) => void;
-    createdOn: string | number;
+    createdAt: string | number;
```
Comment (samshara) [inline src/views/AnalysisModule/Analysis/AnalysisPillar/index.tsx:26]: is createdAt really string | number ?

**Pull request merged**